src/dmd/mars.d

```diff
@@ -464,8 +463,6 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
     if (global.errors)
         removeHdrFilesAndFail(params, modules);
 
-    backend_init();
```

Comment:
I assumed that semantic depended on this somewhere. Maybe typinf or any other of the gluey frontend modules.

Reply:
Hmm, yes. It shouldn't though

Reply:
So it seems that at least semantic1 requires the backend to be initialised: https://github.com/dlang/dmd/pull/12857 any ideas?

Reply:
I could always hoist `backend_init` back out of the new code generation function, but there is something very wrong about semantic analysis requiring an initialised backend.

Reply:
@thewilsonator I changed it to a pointer in [this test](https://github.com/ibuclaw/dmd/commit/c30e57b2637d8fb77703ad00ba8d4e55232f1e4c), and compiler segfaults in `dmd.iasmdmd`.

Reply:
Neat, what were the first few function in the backtrace?

Reply:
Oh, there's only one occurrence nvm

Reply:
@thewilsonator there's more than one occurrence.  See the use of `CodeBuilder` in iasmdmd, and all of its methods.

Reply:
After swapping `config.target_cpu` for `target.cpu`, backtrace is now:
```
#0  0x0000555555b49717 in simplify_code(code*) (c=0x555558d62d78) at src/dmd/backend/cod3.d:6360                                                [1197/1347]
#1  0x0000555555a94dad in CodeBuilder::gen(code*) (this=..., cs=0x7ffffffef490) at src/dmd/backend/codebuilder.d:125
#2  0x0000555555a952b5 in CodeBuilder::genlinnum(Srcpos) (this=..., srcpos=...) at src/dmd/backend/codebuilder.d:304
#3  0x0000555555a6a466 in dmd.iasmdmd.asm_emit(dmd.globals.Loc, uint, dmd.backend.iasm.PTRNTAB, dmd.backend.iasm.OP*, dmd.iasmdmd.OPND[]).finalizeCode(dmd
#4  0x0000555555a6a074 in dmd.iasmdmd.asm_emit(dmd.globals.Loc, uint, dmd.backend.iasm.PTRNTAB, dmd.backend.iasm.OP*, dmd.iasmdmd.OPND[]) (opnds=..., pop=
#5  0x0000555555a65b13 in dmd.iasmdmd.inlineAsmSemantic(dmd.statement.InlineAsmStatement, dmd.dscope.Scope*) (sc=0x7fffe4f4e060, s=0x7fffe4f676a0) at src/
#6  0x0000555555a65591 in asmSemantic(AsmStatement*, Scope*) (s=0x7fffe4f67040, sc=0x7fffe4f4e060) at src/dmd/iasm.d:47
#7  0x0000555555a215b9 in StatementSemanticVisitor::visit(AsmStatement*) (this=0x7fffffff0148, s=0x7fffe4f67040) at src/dmd/statementsem.d:4265
...
```

Reply:
So, IMO, s2ir should be doing the actual code generation.  The semantic pass should just be verifying that all parts of the statement are correct.

Reply:
Yes. Will fix that separately.

Reply:
Don't let me stop you from doing that though!

Reply:
Nah... :-)